config/toml.go

```diff
@@ -0,0 +1,64 @@
+package config
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/BurntSushi/toml"
+)
+
+// RollkitToml is the filename for the rollkit configuration file.
+const RollkitToml = "rollkit.toml"
+
+// TomlConfig is the configuration read from rollkit.toml
+type TomlConfig struct {
+	Entrypoint string          `toml:"entrypoint"`
+	Chain      ChainTomlConfig `toml:"chain"`
+
+	RootDir string
+}
+
+// ChainTomlConfig is the configuration for the chain section of rollkit.toml
+type ChainTomlConfig struct {
+	ConfigDir string `toml:"config_dir"`
+}
+
+// ReadToml reads the TOML configuration from the rollkit.toml file and returns the parsed TomlConfig.
+func ReadToml() (TomlConfig, error) {
+	var config TomlConfig
+	startDir, err := os.Getwd()
+	if err != nil {
+		return config, fmt.Errorf("error getting current directory: %w", err)
+	}
+
+	configPath, err := findConfigFile(startDir)
+	if err != nil {
+		return config, err
+	}
+
+	if _, err := toml.DecodeFile(configPath, &config); err != nil {
+		return config, fmt.Errorf("error reading %s: %w", configPath, err)
+	}
+
+	config.RootDir = filepath.Dir(configPath)
+
+	return config, nil
+}
+
+func findConfigFile(startDir string) (string, error) {
+	dir := startDir
+	for {
+		configPath := filepath.Join(dir, RollkitToml)
+		if _, err := os.Stat(configPath); err == nil {
+			return configPath, nil
+		}
+
+		parentDir := filepath.Dir(dir)
+		if parentDir == dir {
+			break
+		}
+		dir = parentDir
+	}
+	return "", fmt.Errorf("no %s found", RollkitToml)
+}
```

Comment:
We should unit test this file to ensure it operates as intended and also add a comment explaining how the traversal happens. 

For example, reading this I assume you would want to provide a relative path i.e. `appname/config` then the config and appname directories would be checked for the rollkit.toml file. 

In that case, we should warn against passing in the absolute path, since then we are checking all the way up to the root level of the file system, which I don't know if that makes sense. 

Since this function is also only called from `ReadmToml` I believe we are either just checking a single directory, or we are checking recursively all the way up to the root level of the filesystem. 


So to summarize:

1. The scope of this function should be documented in a comment so it is clear the intended action
2. Have a test that validates some of the use cases

Reply:
i'm still on it